src/serializer.test.ts

```diff
@@ -0,0 +1,472 @@
+    expect(link.children[0].text).toBe('with a link');
+  });
+
+  it('deserializes document with <br> tag', () => {
+    const deserialized = deserializeHtmlString(
+      '<p>paragraph <br> with a line break <br/> or two</p>'
+    );
+
+    expect(deserialized).toHaveLength(1);
+    expect(deserialized[0].type).toBe('paragraph');
+    expect(deserialized[0].children).toHaveLength(1);
+    expect(deserialized[0].children[0].text).toBe(
+      'paragraph \n with a line break \n or two'
+    );
+  });
+
+  it('deserializes text with multiple marks', () => {
+    const deserialized = deserializeHtmlString(
+      '<p>this <em>text <strong>is <u>marked <code>up!</code></u></strong></em></p>'
+    );
+
+    expect(deserialized).toHaveLength(1);
+    const paragraph = deserialized[0];
+
+    expect(paragraph.type).toBe('paragraph');
+    expect(paragraph.children).toHaveLength(5);
+
+    expect(paragraph.children[0].text).toBe('this ');
+    checkMarks(paragraph.children[0], {});
+
+    expect(paragraph.children[1].text).toBe('text ');
+    checkMarks(paragraph.children[1], { italic: true });
+
+    expect(paragraph.children[2].text).toBe('is ');
+    checkMarks(paragraph.children[2], { italic: true, bold: true });
+
+    expect(paragraph.children[3].text).toBe('marked ');
+    checkMarks(paragraph.children[3], {
+      italic: true,
+      bold: true,
+      underline: true,
+    });
+
+    expect(paragraph.children[4].text).toBe('up!');
+    checkMarks(paragraph.children[4], {
+      italic: true,
+      bold: true,
+      underline: true,
+      code: true,
+    });
+  });
+
+  it('handles mark tags around other elements', () => {
+    const deserialized = deserializeHtmlString(
+      '<strong><u><h1>Strong header</h1></u><p>and paragraph</p></strong>'
+    );
+
+    expect(deserialized).toHaveLength(2);
+
+    expect(deserialized[0].type).toBe('h1');
+    expect(deserialized[0].children).toHaveLength(1);
+    expect(deserialized[0].children[0].text).toBe('Strong header');
+    checkMarks(deserialized[0].children[0], { bold: true, underline: true });
+
+    expect(deserialized[1].type).toBe('paragraph');
+    expect(deserialized[1].children).toHaveLength(1);
+    expect(deserialized[1].children[0].text).toBe('and paragraph');
+    checkMarks(deserialized[1].children[0], { bold: true });
+  });
+
+  it('ignores unknown tag but finds children', () => {
+    const deserialized = deserializeHtmlString(
+      '<testing-tag><p>should be <taggy-mctagface>found</taggy-mctagface></p></testing-tag>'
+    );
+
+    expect(deserialized).toHaveLength(1);
+
+    expect(deserialized[0].type).toBe('paragraph');
+    expect(deserialized[0].children).toHaveLength(1);
+    expect(deserialized[0].children[0].text).toBe('should be found');
+  });
+
+  it('deserializes an image figure', () => {
+    const deserialized = deserializeHtmlString(
+      '<figure><img src="image_src.jpg" alt="Cool figure 😎"><figcaption>Fig.1 - Cool stuff.</figcaption></figure>'
```

Comment:
Missing a closing slash on the `img` here as well. But, I just realized these were optional in HTML5 on such void elements. Oh well ...